qiskit_ibm/ibm_provider.py

```diff
@@ -757,16 +584,57 @@ def backends(
         """
         # pylint: disable=arguments-differ
         return self._backend.backends(name=name, filters=filters, min_num_qubits=min_num_qubits,
-                                      input_allowed=input_allowed, **kwargs)
+                                      input_allowed=input_allowed, hub=hub, group=group,
+                                      project=project, **kwargs)
```

Comment:
Since you're just calling backend service to get all the backends, why do you need to initialize a `self._backends` here? As is `IBMProvider` calls `BackendService` to get all backends, which then gets all the backends from `IBMProvider`. Why not just store `_backends` in backend service?